presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -116,6 +123,13 @@ public Response killQuery(@PathParam("queryId") QueryId queryId, String message)
         }
     }
 
+    @PUT
+    @Path("{queryId}/preempted")
+    public Response preemptQuery(@PathParam("queryId") QueryId queryId, String message)
```

Comment:
Move this method above, so that it's directly after `killQuery`. That will make it easier to read, since the long helper method will no longer be splitting them up.